packages/service-library/tests/test_json_serialization.py

```diff
@@ -5,38 +5,50 @@
 
 import json
 from typing import Any
-from uuid import UUID, uuid4
+from uuid import uuid4
 
-from servicelib.json_serialization import json_dumps
+import pytest
+from models_library.utils.fastapi_encoders import jsonable_encoder
+from servicelib.json_serialization import OrJsonAdapter, json_dumps
 
-#
 
+def test_json_dump_variants():
 
-def export_uuids_to_str(n: Any):
-    if isinstance(n, dict):
-        for k, v in n.items():
-            n.update({k: export_uuids_to_str(v)})
-    elif isinstance(n, list):
-        n = [export_uuids_to_str(v) for v in n]
-    elif isinstance(n, UUID):
-        return str(n)
-    return n
+    uuid_obj = uuid4()
+
+    with pytest.raises(TypeError) as exc_info:
+        json.dumps(uuid_obj)
+
+    assert str(exc_info.value) == "Object of type UUID is not JSON serializable"
+
+    assert json_dumps(uuid_obj) == json.dumps(str(uuid_obj))
 
 
 def test_serialization_of_uuids(fake_data_dict: dict[str, Any]):
+    # NOTE: UUIDS serialization/deserialization is asymetric.
+    # We should eventually fix this but adding a corresponding decoder?
 
     uuid_obj = uuid4()
     # NOTE the quotes around expected value
     assert json_dumps(uuid_obj) == f'"{uuid_obj}"'
 
     obj = {"ids": [uuid4() for _ in range(3)]}
     dump = json_dumps(obj)
-    assert json.loads(dump) == export_uuids_to_str(obj)
+
+    # NOTE: UUIDs are deserialized as strings, therefore we need to use jsonable_encoder
```

Comment:
the comment is not clear to me. in both situation they are stings and they produce the same outcome, why should we need to use jsonable_encoder?

Reply:
The issues here are `UUID` instances. IMO this is a defect in our design but changing it right now would require a careful PR that I cannot do right now. This test actually is trying to highlight that defect:  `test_serialization_of_uuids`

Reply:
`jsonable_encoder` will automatically turn `UUID`s into `str`.

Reply:
Yes, this is what I already understood. The comment still sounds a bit off to me. It's more confusing having it than not having it.

In my mind saying that you need to use `jsonable_encoder` means that this is the solution to the issue that UUIDs get turned into  strings, but that is not the reason why we use it. We deserialise python objects with it.

Or am I getting this wrong and confusing myself?

Reply:
I am using `jsonable_encoder` to prepare my reference data for comparison. We need this in the test only. just to do the comparison i.e. `==` . 

```python
assert json.loads(dump) == obj
```
will not work because when we serialized the UUIDs of `obj` into `dump` that information got lost. therefore the only way to compare the de-serialized (i.e. `loads`) value is to do
```python
assert json.loads(dump) == jsonable_encoder(obj)
```